zc_common/remote_resource/request.py

```diff
@@ -56,29 +76,51 @@ def get_route_from_fk(resource_type, pk=None):
                 expanded = expand(route, {'id': pk})
             return '{0}{1}'.format(routes[route]['domain'], expanded)
 
-    raise Exception('No route for resource_type: "{0}"'.format(resource_type))
+    raise RouteNotFoundException('No route for resource_type: "{0}"'.format(resource_type))
 
 
-def make_service_request(service_name, endpoint):
-    """Makes a JWT authenticated service request to the URL provided and returns the response.
-    Returns a dictionary of the returned response.
+def make_service_request(service_name, endpoint, method=GET, data=None):
     """
+    Makes a JWT token-authenticated service request to the URL provided.
+
+    Args:
+        service_name: name of the service making this request. e.g. mp-users
+        endpoint: the url to use
+        method: HTTP method. supported methods are defined at this module's global variables
+        data: request payload in case we are making a POST request
+
+    Returns: text content of the response
+    """
+
     jwt_token = jwt_encode_handler(service_jwt_payload_handler(service_name))
     headers = {'Authorization': 'JWT {}'.format(jwt_token), 'Content-Type': 'application/vnd.api+json'}
-    response = requests.get(endpoint, headers=headers)
-    return response.json()
+
+    if not method in [GET, POST]:
+        raise UnsupportedHTTPMethodException(
+            "Method {0} is not supported. service_name: {1}, endpoint: {2}".format(method, service_name, endpoint))
+
+    response = getattr(requests, method)(endpoint, headers=headers, json=data)
```

Comment:
Is the api for the post method here correct? It seems like instead of the `json` key should be `data`? [here](http://docs.python-requests.org/en/master/user/quickstart/#make-a-request)

maybe it's better to build up the kwargs before, only include params that we want to pass and then pass that object into the method argument. This way we don't overwrite the default values. For instance, get and some posts might not have any data at all, so there is no reason to pass that into the method.